src/core/server/ui_settings/types.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { SavedObjectsClientContract, SavedObjectAttribute } from '../saved_objects/types';
+/**
+ * Client that provides access to the UiSettings stored in elasticsearch.
+ *
+ * @public
+ */
+export interface IUiSettingsClient {
+  /**
+   * Returns registered uiSettings values {@link UiSettingsParams}
+   */
+  getRegistered: () => Readonly<Record<string, UiSettingsParams>>;
```

Comment:
Thanks for renaming this. `getDefaults` did not make sense for what was returned.